### ISSUE TITLE
Erroneous 'missing agenda file' message on creating new journal entry 

### ISSUE THREAD

**Issue by mrvdb** [bastibe/org-journal#279]
On every `org-journal-new-entry` call I get a message saying 

 `Non-existent agenda file ~/dat/org/journal/20200727.org  [R]emove from list or [A]bort?`

When pressing [R] : 

 `File was not in list: ~/dat/org/journal/20200727.org (not removed)`

The file mentioned is typically the last journal file.

I suspect this has to do with the setting ` org-journal-carryover-delete-empty-journal 'always` 

I have a `org-agenda-file-regexp` in place which picks up the journal files from `org-journal-dir`

Restarting Emacs makes the message go away

**Comment by cslux** [comment]
> I suspect this has to do with the setting ` org-journal-carryover-delete-empty-journal 'always`

I doubt it.

Please post your org-journal setup please. Please use the issue template the next time. It makes it much easier to fix bugs.

**Comment by cslux** [comment]
Don't forget to provide the value of `org-agenda-file-regexp`. You could also enable debug on quit (`toggle-debug-on-quit`), and hit `C-g` when the message appears, please provide that output too. Thanks.

**Comment by mrvdb** [comment]
Config: https://gist.github.com/mrvdb/696d94ff7c8daf0bffffd75199c5b3bd

I'll post the debug info later (I restarted Emacs this morning, so error isn't there anymore now)

Emacs 28.0.50
OS: Arch Linux

**Comment by mrvdb** [comment]
Backtrace produced by `debug-on-quit`, (C-g pressed after message appeared)
```
Debugger entered--Lisp error: (quit)
  org-check-agenda-file("/home/mrb/dat/org/journal/20200728.org")
  org-agenda-prepare-buffers(("/home/mrb/dat/org/journal/20200728.org"))
  org-map-entries(#f(compiled-function () #<bytecode 0x1e0c29938002>) "+carryover|+TODO=\"TODO\"")
  org-journal-carryover()
  org-journal-new-entry((4))
  funcall-interactively(org-journal-new-entry (4))
  call-interactively(org-journal-new-entry nil nil)
  command-execute(org-journal-new-entry)

**Comment by cslux** [comment]
Should be fixed with commit af2b9a8.
